evalml/pipelines/components/transformers/preprocessing/stl_decomposer.py

```diff
@@ -385,50 +504,105 @@ def _decompose_target(X, y, fh):
             )
 
         if isinstance(y, pd.Series):
-            result_dfs.append(_decompose_target(X, y, None))
-        elif isinstance(y, pd.DataFrame):
-            for colname in y.columns:
-                result_dfs.append(_decompose_target(X, y[colname], None))
-        return result_dfs
+            y = y.to_frame()
+        series_results = {}
+        # Iterate through each series id
+        for id in y.columns:
```

Comment:
This is so late in the process to realize this, but do we actually need to be calling `_decompose_target` in a loop? Since the decomposer handles multiseries generally, it should be able to handle multiseries here too, we can pass `self.periods` through instead of `period=period`, and switch around the logic to return the data in the format we need? That will prevent us from calling `Decomposer.fit()` too many times, which can be slow

Reply:
I modified this so that `get_trend_dataframe` returns a dictionary for multiseries, but for single series it is still returning a list of dataframes. In the `PolynomialDecomposer`, `get_trend_dataframe` returns a list of dataframes. I think this is because there is some multivariate implementation written here. Since they're both using `plot_decomposition` for single series, I wanted to keep the indexing the same so I left it as a list for `STLDecomposer` for now (even though it can be modified later to just return a dataframe). This is pretty inconsistent so I wanted to see what others think about it and if I should consider changing the return type of `get_trend_dataframe` for the multiseries and/or single series `STLDecomposer`.

Reply:
I updated `get_trend_dataframe()` in the `STLDecomposer` to return `list(pd.DataFrame)` for single series and `dict(list(pd.DataFrame))` for multiseries, but it should be updated in the future to no longer be in a list #4294